building-and-packaging/material/packaging_python_slides.md

```diff
@@ -0,0 +1,294 @@
+    version="<version-number>",
+    author="Your Name",
+    author_email="Your Email",
+    description="A small description",
+    url="package-website-url",
+    package_dir={"": "<directory-name>"},
+    packages=setuptools.find_packages(where="<directory-name>"),
+    python_requires=">=3.6",
+)
+
+```
+
+---
+
+## Why is setup.py not the best way?
+
+- Before doing `python setup.py install` you have to manually search and install dependencies
+- A package installed by `setup.py` needs to be manually maintained and uninstalled if required
+- Requires manual downloading of files from the package website
+
+<span>
+Is there a better way?
+<!-- .element: class="fragment" data-fragment-index="1" --></span>
+
+<span>
+Yes! Use pip!
+<!-- .element: class="fragment" data-fragment-index="2" --></span>
+
+---
+
+## What is pip?
+
+- pip is a package installer to install packages from the [Python Package Index PyPI]((https://pypi.org/))
+- pip is itself a [package](https://pypi.org/project/pip/) which is available on PyPI
+- pip is open-source and is developed on [GitHub](https://github.com/pypa/pip)
+
+---
+
+## Using pip
+
+- When `pip install package-name` is run, pip chooses a distribution file and installs it in the environment
+- Doing `python -m pip install package-name` is basically the same as `pip install package-name`
+- pip tracks metadata to allow for easy uninstallation and updating of packages
+- pip is bundled together with Python 3.x, making it even more easier to use
+- pip can install a package from a source distribution (`.tar.gz`) or a wheel distribution (`.whl`)
+- Do not use `sudo pip install`! Various security issues with doing so. Go for `pip install --user <package-name>`
+
+---
+
+## What is PyPI?
+
+- [PyPI](https://pypi.org/) = **The Python Package Index** is a repository of software developed in the Python community
+- PyPI itself is developed on GitHub through another software called [Warehouse](https://github.com/pypa/warehouse)
+- PyPI has a informative [public dashboard](https://p.datadoghq.com/sb/7dc8b3250-85dcf667bd) to show its activity
+- A major advantage is the active maintenance of PyPI and the packages indexed in it
+- Not to be confused with **PyPA** which is Python Packaging Authority, a working group which maintains projects in Python packaging
+
+---
+
+## Example of PyPI package: fenicsprecice
+
+- Having a look at [fenicsprecice](https://pypi.org/project/fenicsprecice/)
+
+---
+
+## Python Enhancement Proposals (PEPs)
+
+- PEP is an evolving design document which provides information regarding new features of Python, new processes and new environments
+- PEPs typically involve concise technical information, which also acts as standardizations
+- Packaging workflows are also standardized through PEPs. Example:
+    - [PEP 427](https://www.python.org/dev/peps/pep-0427/) introduces the built-package format "wheel"
+- It is always a good idea to look for **PEP** conventions before trying out processes
```

Comment:
```suggestion
- It is always a good idea to look for [PEP](https://www.python.org/dev/peps/) conventions before trying out processes
```
Why not directly link on the PEPs? This also gives a visual separation/highlight in the slides while making it easy to obtain more information.